Services/AssessmentQuestion/examples/class.exTestPlayerGUI.php

```diff
@@ -0,0 +1,196 @@
+			$playerQstPageHTML .= $specificFeedbackRenderer->getContent();
+		}
+		
+		/**
+		 * best solution presentation to be answered by the examine
+		 */
+		
+		if( $showBestSolution = true )
+		{
+			$renderer = $questionPresentationGUI->getSolutionPresentation(
+				$questionInstance->getBestSolution()
+			);
+			
+			$playerQstPageHTML .= $renderer->getContent();
+		}
+		
+		$playerQstPageHTML; // complete question page html
+	}
+	
+	/**
+	 * With the presentation of an assessment question, this question also gets submitted having any solution
+	 * filled out by any user. With the first presentation there should be no previous user response available.
+	 * The consuming component needs to request an empty ilAsqSolution instance for the given questionId.
+	 * 
+	 * The ilAsqSolution interface method initFromServerRequest is to be used to initialize the object instance
+	 * with the user response. With the current concept the newly introduced \Psr\Http\Message\ServerRequestInterface
+	 * needs to be injected to this method, but may simply passing $_POST could be an alternative.
+	 * This depends on the future strategy of abstracting the http server request in ILIAS.
+	 * 
+	 * After having this solution saved, the consuming component needs to register the now available solutionId
+	 * together with the questionId and the userId. Additionally this ilAsqSolution instance can be used
+	 * with a question corresponding ilAsqResultCalculator to retrieve the information about right/wrong
+	 * (used for e.g. answer status in CTM's test sequence) and reached points (used as a future ilTestResult)
+	 * to be stored as any result within the consuming component.
+	 * 
+	 * After the first submission of any user response the consuming component needs to provide the corresponsing
+	 * solutionId to request the existing ilAsqSolution instance from the ilAsqFactory for every additional submit.
+	 * 
+	 * The way harvesting and handling solution data in short:
+	 * - post submit gets parsed by ilAsqSolution
+	 * - ilAsqResultCalculator calculates points and right/wrong
+	 * - the test object can use these information for different purposes
+	 * - points can be saved as an ilTestResult referenced by the questionId and the participantId
+	 * - right/wrong can be used for determining the correct feedbacks for the feedback loop
+	 * - right/wrong can be used as the answer status within the CTM test sequence
+	 */
+	public function submitSolution()
+	{
+		global $DIC; /* @var ILIAS\DI\Container $DIC */
+		
+		// this can also be $_REQUEST or any other future ilias post-request handler
+		$serverRequestObject; /* @var \Psr\Http\Message\ServerRequestInterface $serverRequestObject */
+		
+		$questionId = 0; // initialise with id of question that just submits
+		
+		/**
+		 * fetch possibly existing participant solution, an empty one is required otherwise
+		 */
+		
+		$participantSolution = $this->getParticipantSolution($questionId);
```

Comment:
Please consider keeping the existing solution while saving new input. Maybe we need a "saveAsNew"-function and a function which serves an array of the answer-records for a specific participant/test/question.